### PR TITLE
Refactor AlertBox

### PR DESCRIPTION
**Pull request by lubej**
Introduce content prop on AlertBox for existing styling/formatting. Replace children prop, to support custom formatting.

### CODE REVIEW

File: src/app/components/AlertBox/index.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ interface Props {
   center?: boolean
   /** Example `icon={<Info size="20px" />}` */
   icon?: React.ReactNode
-  children: React.ReactNode
+  content: React.ReactNode
```
Comment (buberdds) [inline src/app/components/AlertBox/index.tsx:15]: This prop should be now optional

File: src/app/components/AlertBox/index.tsx [hunk 2]
```diff
@@ -61,12 +61,16 @@ export function AlertBox(props: Props) {
       background={{ color: mapStatus[props.status].background }}
       pad={{ horizontal: 'small' }}
     >
-      <Box direction="row" gap="small" align="center" justify={props.center ? 'center' : 'start'}>
-        {props.icon && <Text color={color}>{props.icon}</Text>}
-        <Text weight="bold" size="12px" style={{ marginTop: 10, marginBottom: 10 }}>
-          {props.children}
-        </Text>
-      </Box>
+      {!props.children && (props.icon || props.content) && (
```
Comment (buberdds) [inline src/app/components/AlertBox/index.tsx:64]: We should remove props.icon from this condition as we allow to render alert box with icon only which is useless.

File: src/app/components/AlertBox/index.tsx [hunk 3]
```diff
@@ -61,12 +61,16 @@ export function AlertBox(props: Props) {
...
+        <Box direction="row" gap="small" align="center" justify={props.center ? 'center' : 'start'}>
+          {props.icon && <Text color={color}>{props.icon}</Text>}
+          <Text weight="bold" size="12px" style={{ marginTop: 10, marginBottom: 10 }}>
+            {props.content}
+          </Text>
+        </Box>
+      )}
+      {/* eslint-disable-next-line no-restricted-syntax */}
```
Comment (buberdds) [inline src/app/components/AlertBox/index.tsx:72]: Why are you disabling lint rule here? you can pass children as a text node <>Foo</> and Google Translate won't work, right?

File: src/app/components/AlertBox/index.tsx (hunk 2 above)
Comment (buberdds) [inline src/app/components/AlertBox/index.tsx:64]: Would be useful to indicate this relation between content and children via types like 
```
type Props = {
  center?: boolean
  status: AlertBoxStatus
} & (
  | {
      content: React.ReactNode
      icon?: React.ReactNode
      children?: undefined
    }
  | {
      children: React.ReactNode
      content?: undefined
      icon?: undefined
    }
)
```

File: src/app/components/AlertBox/index.tsx (hunk 3 above)
Comment (lubej) [reply src/app/components/AlertBox/index.tsx:80]: These are custom rules, it does not work nicely with "dynamic" content. I marked children as `ReactElement | Iterable<React.ReactNode>`, so no direct `string` is allowed. As for why, Google translate changes the structure of the elements, and that is the reason why text should be wrapped inside some element, to not break layout. As far as I know Google Translate will still work.

File: src/app/components/AlertBox/index.tsx [hunk 4]
```diff
@@ -61,12 +69,16 @@ export function AlertBox(props: Props) {
       background={{ color: mapStatus[props.status].background }}
       pad={{ horizontal: 'small' }}
     >
-      <Box direction="row" gap="small" align="center" justify={props.center ? 'center' : 'start'}>
-        {props.icon && <Text color={color}>{props.icon}</Text>}
-        <Text weight="bold" size="12px" style={{ marginTop: 10, marginBottom: 10 }}>
-          {props.children}
-        </Text>
-      </Box>
+      {!props.children && props.content && (
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/AlertBox/index.tsx:72]: With pending txs I was trying to do a bit more "custom" AlertBox:
https://github.com/oasisprotocol/oasis-wallet-web/blob/62c1711b8b07c90b7f386e6081ba667086048f9a/src/app/pages/AccountPage/Features/TransactionHistory/index.tsx#L61-L88

[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/AlertBox/index.tsx:72]: But yes, I agree on `content` naming. Was thinking of naming it `label` first, but that felt counterintuitive as well, as the `label` doesn't have to be only text.

File: src/app/components/AlertBox/index.tsx (hunk 3 above)
Comment (buberdds) [reply src/app/components/AlertBox/index.tsx:80]: why we can't just write  `{props.children}`. Why we need ` {props.children && <>{props.children}</>}` ?

Comment (lubej) [reply src/app/components/AlertBox/index.tsx:80]: Yeah, good point. Now that props are "safeguarded" by TypeScript, should be fine that way.

File: src/app/components/AlertBox/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/AlertBox/index.tsx:72]: localhost_3000_account_oasis1qrfwkqg4n3fd3xs8msljdhyuy0rr34289vcntjft

**Comment by lubej** [comment]
Closing in favor of https://github.com/oasisprotocol/oasis-wallet-web/pull/1978

**Pull request closed without merging**